File: apps/mobile/lib/feature/details/model/seed.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'seed.freezed.dart';
part 'seed.g.dart';

/// Seed data class
@freezed
abstract class Seed with _$Seed {
  const factory Seed({required String guid, required String title, required String externalLink, required int tmdbId}) =
      _Seed;

  const Seed._();

  /// Generate Seed class from Map<String, Object?>
  factory Seed.fromJson(Map<String, Object?> json) => _$SeedFromJson(json);
}

File: apps/stream/src/main/java/org/film/parser/feature/torrent/data/Seed.java
package org.film.parser.feature.torrent.data;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Seed {

    private long tmdbId;
    private String guid;
    private String title;
    private String externalLink;
}
